opencog/attention/WAImportanceDiffusionAgent.h

```diff
@@ -55,14 +55,18 @@ namespace opencog
  * where an extensive explanation of the algorithm, features and pending
  * work is explained.
  */
+class AtomSpace;
 class WAImportanceDiffusionAgent : public ImportanceDiffusionBase
 {
 private:
     unsigned int SAMPLE_SIZE = 1;
     unsigned int _tournamentSize;
+    float  _decayRate; // percentage decay
+    AtomSpace*  _as;
```

Comment:
Anything that inherits from `class Agent` already has an `AtomSpace* _as;` in it see line 126 of `Agent.h`  -- since you've inherited from `ImportanceDiffusionBase` which nherits from `Agent`, you don't need this at all. Its already there.